src/main/kotlin/validation/ValidationHelper.kt

```diff
@@ -50,3 +52,77 @@ fun Raise<ValidationIssue>.ensureEnumNotUnspecified(name: String, value: Enum<*>
     ensure(runCatching { UUID.fromString(id) }.isSuccess) { InvalidId(name, id) }
+
+/**
+ * Ensures that the provided email has a valid format.
+ * If the email does not match the [EMAIL_REGEX], an [InvalidEmail] validation issue is raised.
+ *
+ * @param email The email address to validate.
+ */
+fun Raise<ValidationIssue>.ensureEmailValidity(email: String) =
+    ensure(EMAIL_REGEX.matches(email)) { InvalidEmail(email) }
+
+/**
+ * Ensures that the provided password meets the required complexity criteria.
+ * If the password does not meet the criteria, an [InvalidPassword] validation issue is raised.
+ *
+ * The criteria for a valid password are:
+ * - At least 8 characters long
+ * - At least 2 lowercase letters
+ * - At least 2 uppercase letters
+ * - At least 2 digits
+ * - At least 2 special characters (defined by [SPECIAL_CHAR_REGEX])
+ *
+ * @param password The password to validate.
+ */
+fun Raise<ValidationIssue>.ensurePasswordValidity(password: String) {
+    ensure(password.length >= PASSWORD_MIN_LENGTH) {
+        InvalidPassword(password, "must be at least $PASSWORD_MIN_LENGTH characters long")
+    }
+    ensure(password.count { it.isLowerCase() } >= PASSWORD_MIN_NUMBER_LOWERCASE_LETTERS) {
+        InvalidPassword(
+            password,
+            "must contain at least $PASSWORD_MIN_NUMBER_LOWERCASE_LETTERS lowercase letters",
+        )
+    }
+    ensure(password.count { it.isUpperCase() } >= PASSWORD_MIN_NUMBER_UPPERCASE_LETTERS) {
+        InvalidPassword(
+            password,
+            "must contain at least $PASSWORD_MIN_NUMBER_UPPERCASE_LETTERS uppercase letters",
+        )
+    }
+    ensure(password.count { it.isDigit() } >= PASSWORD_MIN_NUMBER_DIGITS) {
+        InvalidPassword(
+            password,
+            "must contain at least $PASSWORD_MIN_NUMBER_DIGITS digits",
+        )
+    }
+    ensure(SPECIAL_CHAR_REGEX.findAll(password).count() >= PASSWORD_MIN_NUMBER_SPECIAL_CHARS) {
+        InvalidPassword(
+            password,
+            "must contain at least $PASSWORD_MIN_NUMBER_SPECIAL_CHARS special characters",
+        )
+    }
+}
+
+/**
+ * Ensures that the provided first name is valid.
+ * It checks that the first name is not blank and does not exceed the maximum length defined by [FIRST_NAME_MAX_LENGTH].
+ *
+ * @param firstName The first name to validate.
+ */
+fun Raise<ValidationIssue>.ensureFirstNameValidity(firstName: String) {
+    ensureFieldNonBlank("first name", firstName)
+    ensureFieldLength("first name", firstName, FIRST_NAME_MAX_LENGTH)
+}
+
+/**
+ * Ensures that the provided last name is valid.
+ * It checks that the last name is not blank and does not exceed the maximum length defined by [LAST_NAME_MAX_LENGTH].
+ *
+ * @param lastName The last name to validate.
+ */
+fun Raise<ValidationIssue>.ensureLastNameValidity(lastName: String) {
+    ensureFieldNonBlank("last name", lastName)
+    ensureFieldLength("last name", lastName, LAST_NAME_MAX_LENGTH)
```

Comment:
Here we want to use the name of the field so that the user gets feedback if they misspelled the field.
```suggestion
fun Raise<ValidationIssue>.ensureFirstNameValidity(firstName: String) {
    ensureFieldNonBlank("first_name", firstName)
    ensureFieldLength("first_name", firstName, FIRST_NAME_MAX_LENGTH)
}

/**
 * Ensures that the provided last name is valid.
 * It checks that the last name is not blank and does not exceed the maximum length defined by [LAST_NAME_MAX_LENGTH].
 *
 * @param lastName The last name to validate.
 */
fun Raise<ValidationIssue>.ensureLastNameValidity(lastName: String) {
    ensureFieldNonBlank("last_name", lastName)
    ensureFieldLength("last_name", lastName, LAST_NAME_MAX_LENGTH)
```